### ISSUE TITLE
Footnote 66 explaining the first ! (bang)

### ISSUE THREAD

**Issue by marcbezem** [UniMath/SymmetryBook#175]
The first use of ! (the Big Bang of the bang, so to say) occurs on p. 44 and is explained in footnote 66:

"Elements of a type that is a proposition carry no information beyond witnessing that the proposition is non-empty, since all elements of a proposition can be identified. Therefore we may replace them by an exclamation mark."

 Should we leave out "carry no information beyond witnessing that the proposition is non-empty"? After all, such elements can carry information, such as about a center of contraction, or the number of elements in a finite type. It is true that all other elements of the type that is a proposition carry the same information, but that is something else than carrying no information.

Also, this use of ! is such an important convention that it deserves a more prominent announcement, perhaps already in 2.15?

**Comment by DanGrayson** [comment]
Yes, a witness to contractibility provides the center of contraction.  But then, any element of the contractible type is the center of a contraction.  So in some sense, it carries no information.  Well, it provides an element of the type, which might not be available otherwise.

(Being a finite type is not a proposition.)

**Comment by marcbezem** [comment]
I was thinking of Lemma 2.24.4(1), concerning finite types.

**Comment by bidundas** [comment]
I’m in luminy listening to a torrential rognes lecture and can’t come to the book meeting.

Bjorn

On 23 Jan 2023, at 21:44, Marc Bezem ***@***.***> wrote:

﻿

I was thinking of Lemma 2.24.4(1), concerning finite types.

**Comment by marcbezem** [comment]
Progress report: I removed "!" from Ch. 2 (just a few) and have moved on to Ch. 3, looking for a good place to introduce this (very convenient) convention. When I have found that place I'll let you know and then the issue can be closed if you are all happy with my solution.
